source/GameInterface/Services/Clans/Patches/ClanHeirSelectionPatch.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Common;
+using Common.Messaging;
+using GameInterface.Services.Clans.Messages;
+using HarmonyLib;
+using Helpers;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Actions;
+using TaleWorlds.CampaignSystem.Extensions;
+using TaleWorlds.CampaignSystem.LogEntries;
+using TaleWorlds.CampaignSystem.Party.PartyComponents;
+using TaleWorlds.CampaignSystem.Roster;
+using TaleWorlds.CampaignSystem.Settlements;
+using TaleWorlds.Core;
+using TaleWorlds.Localization;
+
+namespace GameInterface.Services.Clans.Patches
+{
+    [HarmonyPatch(typeof(ApplyHeirSelectionAction), "ApplyInternal")]
+    public class ClanHeirSelectionPatch
+    {
+        public static bool Prefix(Hero heir, bool isRetirement = false)
+        {
+            string playerHeroId = Hero.MainHero.StringId;
+
+            MessageBroker.Instance.Publish(heir, new LocalNewHeir(heir.StringId, playerHeroId, isRetirement));
+
+            return false;
+        }
+
+        public static void RunFixedHeirSelectionPatchHero(Hero heir, Hero playerHero, bool isRetirement = false)
```

Comment:
What does this function do exactly?

Reply:
The HeirSelection ApplyInternal contains a lot of references to Clan.PlayerClan and MobileParty.MainParty, this removes all of those and runs it using the correct people.

An option to using transpilers?